### ISSUE TITLE
[TF Research Step 13 Execution] Run preregistered learned-slot remaining-prior P0-P9 validation probe

### ISSUE THREAD

**Issue by Irthn1311** [Irthn1311/FER2013_Graph#40]
## Objective
Execute exactly one preregistered **validation-only, fixed-checkpoint, inference-only** scientific experiment using the Step 13 remaining-prior decomposition harness merged from PR #39.

Research question:

> After replacing fixed anatomical local residual pooling with learned local residual slots, which remaining explicit MediaPipe / semantic-prior pathways still carry HIGH functional sensitivity at the selected seed-42 checkpoint?

This issue is the **scientific execution step**, not a new architecture experiment and not a mitigation experiment.

## Exact source state

Run from exact `main` merge commit:

`d90cce8c4d23f8f1c2958c76cda4ce9d8cae6608`

The implementation review PASS was tied to PR #39 head:

`a0e10465c7ea352f3b20f4ad1f8405b55f2d2fca`

Canonical Step 13 harness identities after blocker closure:

- probe SHA-256: `cf68c47d428d0b569828d65028024fcc0713e963419ff5511be91b1377327118`
- harness evidence SHA-256: `09e2e999f4e3787f691d33ea4eeaf2ded1788e63e072dfd5c860e7d125ce4098`

No code, threshold, checkpoint, model, Q, condition semantics, dataset split, or frozen package identity may change for this run.

## Locked scientific identities

Dataset split signature:

`fer2013_train28709_val3589_test3589`

Use **validation split only**: exactly 3,589 validation samples.

Selected Step-12E candidate checkpoint identities:

- Step-12E archive SHA-256: `f436b0a7a20c751b2fd2f47738469fb409ecf9a1a40628e05d20974639927451`
- epoch-42 `.keras`: `e0d633cb6200e963f31a28750e28c7febdaae40344c90ba9d94b826a09e4b78c`
- weights: `a18a372f70ce56868ae43257e9b7fa5e20517499c2c1e35c48dba4d65eaaaa74`
- metadata: `a5ee759bc6fbef587e025199d0dcfe6ebd3a1764cffa567f793c53e972eb47cf`
- learned-slot Q: `54b368aa183c65d5843d8b8e340d3020412d1a2dfeaabbe8b2c0166684ab3ff9`
- resolved config: `3c028dd2f32ebed3a252544e170220b150b5e29920cea865924dddce6aef5a32`
- candidate class: `LearnedLocalResidualSlotLapGNN`
- params: `1,061,576`
- variables: `128`
- Q variable index: `127`
- Q shape/dtype: `[4,96]`, `float32`

Frozen scientific payload SHA-256:

`286be711a53b76511bcf3b9bf949fad694f7c7d272392f9defc56f4914822c0e`

Frozen execution contract:

`14acc2750875a25922007459161a137158d8040805e616166be923f63658bf22`

## Conditions — exact P0–P9

Topology, graph membership, node/edge counts, coordinates, checkpoint weights, Q, and all unregistered features must remain fixed.

- **P0 — official candidate manual forward**: no intervention.
- **P1 — face mask off**: zero only node feature `[:,5]`.
- **P2 — node part-soft off**: zero only node features `[:,6:19]`; direct `part_soft` interfaces remain official.
- **P3 — distance maps off**: zero only node features `[:,19:31]`.
- **P4 — landmark-missing flag off**: zero only node feature `[:,31]`.
- **P5 — semantic edge features off**: zero only edge features `[:,6:8]`.
- **P6 — GNN context part-soft off**: zero only the `part_soft` passed into `gnn.encode`; downstream `part_pool` and readout part-soft remain official; learned slots are recomputed from the altered post-GNN `h`.
- **P7 — readout part-soft off**: zero only the `part_soft` passed into readout; upstream remains official.
- **P8 — local readout validity off**: upstream/readout part-soft remain official; set only the four local readout validity flags to false while global remains true.
- **P9 — joint explicit-prior anchor**: zero node `[:,5:32]`, zero edge `[:,6:8]`, zero context part-soft, zero readout part-soft, and set the four local readout validity flags false/global true.

P9 is **not** MediaPipe-free because the official graph topology remains unchanged.

Preserve node visual channels `0:5` and `32:37`, edge channels `0:6`, topology/index arrays, graph membership, sample ordering, Q, model weights, and all non-registered interfaces.

## Gate A — P0 implementation identity

Compare native candidate forward vs P0 manual forward over the full validation split.

Required PASS:

- prediction/probability agreement: `1.0`
- max absolute logits difference: `<= 1e-5`
- max absolute probability difference: `<= 3e-6`

Any failure => **INVALID_RUN / STOP**. Do not interpret P1–P9.

## Gate B — selected checkpoint reproduction

P0 must reproduce the selected epoch-42 Step-12E checkpoint on exactly 3,589 validation samples:

- accuracy reference: `0.6232933964892727`, tolerance `±0.001`
- macro-F1 reference: `0.596090717851928`, tolerance `±0.001`
- loss reference: `1.1486882999934982`, tolerance `±0.005`

Any failure => **INVALID_RUN / STOP**. Do not interpret P1–P9.

## Gate C — provenance / identity

Before scientific interpretation, all locked source/config/checkpoint/model/Q/frozen-package identities must PASS, including exact resolved-config SHA and metadata/config provenance cross-check for:

- canonical config hash
- seed
- package checksum
- execution contract
- graph signature
- feature signature
- prior signature
- dataset split signature

The implementation must also verify source/config/model artifacts remain unchanged after evaluation.

Any failure => **INVALID_RUN / STOP**.

## Primary sensitivity endpoint

For each `Pi`, `i=1..8`:

`delta_macro_f1_pp(Pi) = 100 * (P0_macro_f1 - Pi_macro_f1)`

Labels are locked **before outcomes**:

- `HIGH` if delta `>= 10.0 pp`
- `MODERATE` if `>= 5.0 pp` and `< 10.0 pp`
- `LOW` if `< 5.0 pp`
- negative delta is also `LOW`

Do not change thresholds after seeing results.

Do not add nonlinear probe deltas together and do not report them as percentage contribution / variance explained / causal share.

## Overall decision rule

Count HIGH labels only among **P1–P8**.

- `>=2 HIGH` => `MULTIPLE_HIGH_REMAINING_PRIOR_PATHS`
- exactly `1 HIGH` => `SINGLE_HIGH_REMAINING_PRIOR_PATH`
- `0 HIGH` and P9 delta `>=10 pp` => `DISTRIBUTED_OR_INTERACTION_REMAINING_PRIOR_DEPENDENCY`
- `0 HIGH` and P9 delta `<10 pp` => `LOW_REMAINING_EXPLICIT_PRIOR_SENSITIVITY`

P9 is an interaction/joint anchor only; it is not counted as an individual pathway.

## Required outputs

Produce a fresh evidence artifact / machine-readable manifest containing at minimum:

1. exact repo/source/checkpoint/config/Q/frozen identities;
2. environment/runtime identity;
3. Gate A/B/C results;
4. P0–P9 sample count, accuracy, macro-F1, loss;
5. P1–P8 delta macro-F1 in pp and locked label;
6. P9 delta macro-F1 in pp;
7. paired prediction disagreement and correctness-transition diagnostics already defined by the reviewed harness;
8. topology/source/model immutability checks;
9. final overall decision label from the locked rule above;
10. explicit statement that test split was not accessed.

Record the resulting evidence artifact SHA-256.

## Runtime

Use the registered Step-12 continuation-compatible environment unless the execution adapter only records equivalent environment identity without changing scientific semantics:

- Python 3.12.12
- TensorFlow 2.18.1
- Keras 3.15.0
- CUDA 12.5.1
- cuDNN 9
- 2 × Tesla T4
- eval batch size 32
- workers 2
- prefetch 2
- cache 64
- mixed precision on
- XLA off
- memory growth on

This is inference-only: no training, optimizer, gradient update, checkpoint/model selection, or augmentation.

## Hard prohibitions

- **Absolutely no test split access** — no labels, predictions, metrics, diagnostics, debugging, or smoke test.
- No training / fine-tuning / optimizer / gradient update.
- No model or checkpoint selection.
- No new mitigation, regularization, or architecture change in this issue.
- No rebuilding topology under interventions.
- Do not call P9 MediaPipe-free.
- Do not claim a HIGH path causes overfitting.
- Do not convert fixed-checkpoint OOD sensitivity into causal contribution percentages.
- Do not treat LOW or NO_CLEAR-type outcomes as statistical equivalence.
- Do not modify the frozen package.

## Scientific interpretation boundary

Allowed language after a valid run:

> “Pathway X shows HIGH/MODERATE/LOW functional sensitivity under the preregistered fixed-checkpoint OOD neutralization.”

Not allowed:

> “Pathway X causes overfitting,” “MediaPipe contributes X%,” or “removing X during training will improve generalization.”

The outcome of this issue determines the next **separate** experiment. No Step 14 mitigation may be selected before this result is reviewed.

## Completion condition

This issue is complete only when:

1. the exact execution adapter/run command, if any new adapter is required, has been implementation-reviewed without changing the registered scientific protocol;
2. the full validation P0–P9 run completes with Gates A/B/C PASS;
3. the generated evidence is independently reviewed against this preregistration;
4. a scientific decision is recorded without touching the test split.

**Comment by Irthn1311** [comment]
## Preregistration amendment — exact execution-wrapper checkout lock

Recorded **before any full Step 13 scientific execution**.

PR #41 (`TF Step 13 registered execution preparation`) passed implementation review at exact head:

`c15f5ec80a33533624012e54d8520ee35bd3a4bf`

It was merged with an exact-head guard. The only authorized execution checkout for the registered Issue #40 full-validation P0–P9 run is now exact `main` commit:

`56a22a3200f1e0dc0100093d43f0db296c11487d`

Scientific base remains unchanged:

`d90cce8c4d23f8f1c2958c76cda4ce9d8cae6608`

Execution-wrapper identity:

- adapter path: `tools/run_issue40_step13_execution.py`
- adapter SHA-256: `2ec514beb399ce5fe62a5cf37989f8317039ca8a2423f79400d1b288e799bec3`
- reviewed scientific probe SHA-256: `cf68c47d428d0b569828d65028024fcc0713e963419ff5511be91b1377327118`
- execution-preparation evidence SHA-256: `d462237bffcfdfe122eab6852f012f768d8a0ce3def6fad8862bd1d7fffee54a`

The wrapper is fail-closed: outer exit `0` requires `status == COMPLETE` and `scientific_result_valid == true`; ordinary probe/Gate/manifest invalidity returns stable exit `3` after preserving available evidence; `KeyboardInterrupt`/unexpected `SystemExit` propagate.

### Authorized run command

```bash
python /kaggle/working/FER2013_Graph/tools/run_issue40_step13_execution.py \
  --repository-root /kaggle/working/FER2013_Graph \
  --step12e-source /kaggle/input/datasets/irthn1311/tf-step12e-reviewed/tf_step12c_checkpoint_continuation_kaggle_t4.zip.b64 \
  --prior-root /kaggle/input/datasets/irthn1311/d16-mediapipe-pixel-priors-best-retry-rescue/outputs/d16_mediapipe_pixel_priors_best_retry_rescue \
  --clean-graph-cache-dir /kaggle/input/datasets/irthn1311/ofix7-mid-seed42-records \
  --output-root /kaggle/working/tf_step13_remaining_prior_probe_run \
  --archive-path /kaggle/working/tf_step13_remaining_prior_probe_kaggle_t4.zip \
  --confirm RUN_ISSUE40_VALIDATION_ONLY_P0_P9
```

Before invoking it, the execution checkout must verify `git rev-parse HEAD == 56a22a3200f1e0dc0100093d43f0db296c11487d` and a clean tracked worktree. No later commit is authorized without another preregistration amendment.

All original Issue #40 P0–P9 semantics, Gate A/B/C thresholds, checkpoint/model/Q/config identities, validation-only boundary, and hard prohibition on test access remain unchanged.

**Execution is now authorized exactly once under this lock.** No Step 14 work is authorized until the resulting evidence archive is independently reviewed.

**Comment by Irthn1311** [comment]
## Research-lead clarification after preflight stop

The previous Step-13 execution did **not** begin. It stopped before the scientific command because the Kaggle workspace was still at PR #41 head `c15f5ec80a33533624012e54d8520ee35bd3a4bf` instead of the already-preregistered post-merge execution checkout `56a22a3200f1e0dc0100093d43f0db296c11487d`.

Therefore:

- this was a **preflight workspace-state failure**, not a scientific P0-P9 attempt;
- no one-run scientific authorization has been consumed;
- no P0-P9 result, Gate outcome, metric, or evidence archive exists from that stop;
- the status should be treated as `STEP13_EXECUTION_NOT_STARTED_PREFLIGHT_STOP`, not as an invalid scientific result.

### Authorized workspace correction

Before the one registered Step-13 attempt, it is now explicitly authorized to bring the Kaggle execution checkout to the already-preregistered exact commit:

`56a22a3200f1e0dc0100093d43f0db296c11487d`

Use an explicit fetch plus detached checkout/reset only for this workspace correction, for example:

```bash
git fetch origin

git checkout --detach 56a22a3200f1e0dc0100093d43f0db296c11487d

git rev-parse HEAD
git status --porcelain
```

Required immediately before execution:

- HEAD exactly `56a22a3200f1e0dc0100093d43f0db296c11487d`;
- tracked worktree clean;
- adapter/probe hashes and all Issue #40 locks pass.

Only after those checks may the previously registered exact scientific command be run **once**.

If checkout/fetch/preflight still fails, stop again without running and return the technical evidence. Do not substitute another commit. Do not patch the scientific code in the execution workspace.

Generation-2 CF-HPG work under Issue #42 is a separate research branch and must not modify or reinterpret this Step-13 protocol.
